src/connector/src/source/iceberg/mod.rs

```diff
@@ -196,49 +199,81 @@ pub enum IcebergTimeTravelInfo {
 }
 
 impl IcebergSplitEnumerator {
-    pub async fn list_splits_batch(
+    pub async fn list_splits_batch_count_star(
```

Comment:
If it is a count star query, we don't need to return multiple splits, i.e. we don't need to run it in a distributed way.